craft_parts/parts.py

```diff
@@ -464,6 +476,13 @@ def _check_partitions_in_filepaths(
                             f"    misused partition {partition!r} in {filepath!r}"
                         )
 
+            if require_inner_path:
+                _, inner_path = get_partition_and_path(filepath)
+                if not inner_path:
+                    error_list.append(
+                        f"    no path specified after partition in {filepath!r}"
```

Comment:
this might be a todo for later, but I would have made the error list about the data and left the formatting to the exception that renders it as a string

Reply:
please create a task for this

Reply:
Created #670